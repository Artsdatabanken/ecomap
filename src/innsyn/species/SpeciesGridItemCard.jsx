import React from 'react'
import classificationLevels from './classificationLevels.js'
import IconButton from 'material-ui/IconButton'
import StarBorder from 'material-ui/svg-icons/toggle/star-border'
import { Card, CardMedia, CardTitle } from 'material-ui/Card'

class SpeciesGridItemCard extends React.Component {
  constructor () {
    super()
    this.state = {}
  }
  render () {
    const tile = this.props.tile
    return (
      <Card
        key={tile.id}
        onClick={() => this.props.onClick(tile)}
        actionIcon={
          <IconButton>
            <StarBorder color='white' />
          </IconButton>
        }
        style={{ width: '100%', marginBottom: 2 }}
        cols={tile.featured ? 2 : 1}
        rows={tile.featured ? 2 : 1}
      >
        <CardMedia
          overlay={<CardTitle
            title={tile.scientificName + ' (' + classificationLevels[tile.level] + ')'}
            subtitle={tile.popularName} />}
        >
          <span
          >
            <img src={tile.imageUrl} alt={'Photo of ' + tile.imageScientificName}
              title={tile.imageScientificName + ' - ' + tile.imageAttribution}
              onLoad={() => this.setState({ imageLoaded: true })}
              style={{
                width: '100%',
                minHeight: 132,
                maxHeight: this.state.imageLoaded ? 2300 : 400,
                opacity: this.state.imageLoaded ? 1 : 0,
                transition: 'opacity 0.5s, max-height 0.5s ease-in'
              }}
            />
          </span>
        </CardMedia>
      </Card>
    )
  }
}

export default SpeciesGridItemCard
