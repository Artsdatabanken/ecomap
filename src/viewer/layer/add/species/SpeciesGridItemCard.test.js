import React from 'react'
import renderer from 'react-test-renderer'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

import SpeciesGridItemCard from './SpeciesGridItemCard'

const loadingTile = {
  id: 2342,
  scientificName: 'Alces Alces',
  level: 4,
  popularName: 'Elg'
}
const loadedFeaturedTile = {
  ...loadingTile,
  imageUrl: 'http://elg.no',
  imageAttribution: 'CCBY',
  imageScientificName: 'xAlces'
}

it('loading standard tile', () => {
  const tree = renderer.create(
    <MuiThemeProvider>
      <SpeciesGridItemCard tile={loadingTile} />
    </MuiThemeProvider>
      )
  expect(tree.toJSON()).toMatchSnapshot()
})

it('complete featured tile', () => {
  const tree = renderer.create(
    <MuiThemeProvider>
      <SpeciesGridItemCard tile={loadedFeaturedTile} />
    </MuiThemeProvider>
      )
  expect(tree.toJSON()).toMatchSnapshot()
})
