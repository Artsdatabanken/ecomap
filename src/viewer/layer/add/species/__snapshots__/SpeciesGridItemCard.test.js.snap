// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complete featured tile 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "marginBottom": 2,
      "muiPrepared": true,
      "pageBreakInside": "avoid",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <div
      style={
        Object {
          "muiPrepared": true,
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <span
          style={
            Object {
              "maxWidth": "100%",
              "minWidth": "100%",
              "muiPrepared": true,
              "verticalAlign": "top",
              "width": "100%",
            }
          }
        >
          <img
            alt="Photo of xAlces"
            onLoad={[Function]}
            src="http://elg.no"
            style={
              Object {
                "maxHeight": 400,
                "minHeight": 132,
                "opacity": 0,
                "transition": "opacity 0.5s, max-height 0.5s ease-in",
                "width": "100%",
              }
            }
            title="xAlces - CCBY"
          />
        </span>
      </div>
      <div
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "muiPrepared": true,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "muiPrepared": true,
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "background": "rgba(0, 0, 0, 0.54)",
                "bottom": 0,
                "left": 0,
                "muiPrepared": true,
                "paddingTop": 8,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <div
              style={
                Object {
                  "muiPrepared": true,
                  "padding": 16,
                  "position": "relative",
                }
              }
            >
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "display": "block",
                    "fontSize": 24,
                    "lineHeight": "36px",
                    "muiPrepared": true,
                  }
                }
              >
                Alces Alces (Rekke)
              </span>
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.54)",
                    "display": "block",
                    "fontSize": 14,
                    "muiPrepared": true,
                  }
                }
              >
                Elg
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`loading standard tile 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "marginBottom": 2,
      "muiPrepared": true,
      "pageBreakInside": "avoid",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <div
      style={
        Object {
          "muiPrepared": true,
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <span
          style={
            Object {
              "maxWidth": "100%",
              "minWidth": "100%",
              "muiPrepared": true,
              "verticalAlign": "top",
              "width": "100%",
            }
          }
        >
          <img
            alt="Photo of undefined"
            onLoad={[Function]}
            src={undefined}
            style={
              Object {
                "maxHeight": 400,
                "minHeight": 132,
                "opacity": 0,
                "transition": "opacity 0.5s, max-height 0.5s ease-in",
                "width": "100%",
              }
            }
            title="undefined - undefined"
          />
        </span>
      </div>
      <div
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "muiPrepared": true,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "muiPrepared": true,
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "background": "rgba(0, 0, 0, 0.54)",
                "bottom": 0,
                "left": 0,
                "muiPrepared": true,
                "paddingTop": 8,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <div
              style={
                Object {
                  "muiPrepared": true,
                  "padding": 16,
                  "position": "relative",
                }
              }
            >
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "display": "block",
                    "fontSize": 24,
                    "lineHeight": "36px",
                    "muiPrepared": true,
                  }
                }
              >
                Alces Alces (Rekke)
              </span>
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.54)",
                    "display": "block",
                    "fontSize": 14,
                    "muiPrepared": true,
                  }
                }
              >
                Elg
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
